xskillscore/tests/test_deterministic.py

```diff
@@ -130,20 +131,108 @@ def test_correlation_metrics_xr(a, b, dim, weight_bool, weights, metrics):
     assert_allclose(actual, expected)
 
 
-@pytest.mark.parametrize("metrics", distance_metrics)
+@pytest.mark.parametrize("metrics", correlation_metrics)
 @pytest.mark.parametrize("dim", AXES)
 @pytest.mark.parametrize("weight_bool", [True, False])
 @pytest.mark.parametrize("skipna", [True, False])
-@pytest.mark.parametrize("has_nan", [True, False])
-def test_distance_metrics_xr(a, b, dim, weight_bool, weights, metrics, skipna, has_nan):
+def test_correlation_metrics_xr_nan(
+    a_nan, b_nan, dim, weight_bool, weights, metrics, skipna
+):
+    """Test whether correlation metric for xarray functions (from
+    deterministic.py) give save numerical results as for numpy functions (from
+    np_deterministic.py) with nans."""
+    a = a_nan
+    b = b_nan
+    # unpack metrics
+    metric, _metric = metrics
+    # Only apply over time dimension for effective p value.
+    if (dim != "time") and (metric in temporal_only_metrics):
+        dim = "time"
+    # Generates subsetted weights to pass in as arg to main function and for
+    # the numpy testing.
+    _weights = adjust_weights(dim, weight_bool, weights)
+    #
+    if metric in temporal_only_metrics:
+        actual = metric(a, b, dim, skipna=skipna)
+    else:
+        actual = metric(a, b, dim, weights=_weights, skipna=skipna)
+    # check that no chunks for no chunk inputs
+    assert actual.chunks is None
+
+    dim, _ = _preprocess_dims(dim, a)
+    if len(dim) > 1:
+        new_dim = "_".join(dim)
+        _a = a.stack(**{new_dim: dim})
+        _b = b.stack(**{new_dim: dim})
+        if weight_bool:
+            _weights = _weights.stack(**{new_dim: dim})
+    else:
+        new_dim = dim[0]
+        _a = a
+        _b = b
+    _weights = _preprocess_weights(_a, dim, new_dim, _weights)
+
+    # ensure _weights.values or None
+    _weights = None if _weights is None else _weights.values
+
+    axis = _a.dims.index(new_dim)
+    if metric in temporal_only_metrics:
+        res = _metric(_a.values, _b.values, axis, skipna=skipna)
+    else:
+        res = _metric(_a.values, _b.values, _weights, axis, skipna=skipna)
+    expected = actual.copy()
+    expected.values = res
+    assert_allclose(actual, expected)
+
+
+@pytest.mark.parametrize("metrics", distance_metrics)
+@pytest.mark.parametrize("dim", AXES)
+@pytest.mark.parametrize("weight_bool", [True, False])
+def test_distance_metrics_xr(a, b, dim, weight_bool, weights, metrics):
```

Comment:
Too me the test names here were all misleading. The docstring reflects what the tests do, but not the titles...